docs/getting_started/DMUpgrade.md

```diff
@@ -0,0 +1,313 @@
+
+## Parameters
+
+
+### Build parameters
+
+| Name               | Description                                   |  Type  | Value     |
+|--------------------|-----------------------------------------------|:------:|:----------|
+| `artifact_repo`    | The git repo url where all current CRDs exist | string | `git_url` |
+| `prev-spec-branch` | Branch of current CRDs to compare to new CRDs | string | `master`  |
+
+
+
+
+### Runtime/Install parameters
+
+
+| Name                                         | Description                                         | Type | Value   |
+|----------------------------------------------|-----------------------------------------------------|:----:|:--------|
+| `force`                                      | Force upgrade of a datamodel                        | bool | `false` |
+| `datamodel_backward_compatibility_validator` | Flag to enable/disable backward compatibility check | bool | `false` |
+
+Specify each parameter using the `--key=value` argument to `nexus datamodel build`. For example,
+
+```
+$ nexus datamodel build --name orgchart  --artifact_repo=<git_repo> --prev_spec_branch=<master>
+```
+
+The above command builds the new datamodel against the master branch of the artifactory repo and force will be false.<br>
+If prev_spec_branch is not provided then force will be set to true.
+
+## What is backward compatibility check?
+The Backward Compatibility Check is a validation process performed during a datamodel upgrade to ensure that the changes made to the DSL are backward compatible with the existing CRDs (Custom Resource Definitions), which serve as the source of truth.
+<br> The check is performed at build time and runtime
+
+<br>At build time :<br>
+* If "prev-spec-branch" is not provided, force will be set to true and build/crds dir will be the source of truth if present.
+* If "prev-spec-branch" is provided and artifact_repo is provided, force will be set to false and source of truth will be the 'crds' dir present in prev_spec_branch of artifact_repo provided
+* If the prev-spec-branch is provided but the artifact_repo is not provided, the nexus.yaml file is checked for the artifact_repo information. If it is not found, an error is thrown, and the process exits. <br><br>
+
+<br>At runtime:<br>
+* CRDs existing in the cluster will be the source of truth 
+
+## When does backward compatibility check fail?
+When the change made to the DSL is determined as backward-incompatible (as defined below) then the backward compatibility check in CI will fail.<br>
+To enable this we should provide prev-spec-branch and artifact_repo info to set force to false as mentioned above.
+
+## How to perform force upgrade at runtime?
+1. Ensure there are no CR objects for the dsl node crd that is being upgraded
+Port forward port 5001 of nexusapp-mgr pod in cosmos-cd namespace
+```
+kubectl port-forward -n cosmos-cd nexus-app-mgr-7865f8df58-h7hbj 5018:5001
```

Comment:
This step is only applicable for Trion (not for Nexus).
So, for Nexus, we should provide the steps to set the force flags via Nexus cli.
For trion, We can capture this detail in a confluence page.